test and document confluent ssl options for sinks (#11634)
MaterializeInc/materialize: doc/user/content/sql/create-sink.md

**TOPIC** _consistency&lowbar;topic_ | Makes the sink emit additional [consistency metadata](#consistency-metadata) to the named topic. Only valid for Kafka sinks. If `reuse_topic` is `true`, a default naming convention will be used when the topic name is not explicitly set. This is formed by appending `-consistency` to the output topic name. {{< version-added v0.8.4 />}}
**AVRO OCF** _path_ | The absolute path and file name of the Avro Object Container file (OCF) to create and write to. The filename will be modified to let Materialize create a unique file each time Materialize starts, but the file extension will not be modified. You can find more details [here](#avro-ocf-sinks).
_sink&lowbar;with&lowbar;options_ | Options affecting sink creation. For more detail, see [`WITH` options](#with-options).

> _with&lowbar;options_ | Options affecting Materialize's connection to Kafka. For more detail, see [SSL and Authentication](#ssl-and-authentication).

Reviewer: ```suggestion
_with&lowbar;options_ | Options affecting Materialize's connection to Kafka. For more detail, see [SSL and Authentication](#authentication).
```

`replication_factor` | `int`      | Set the sink Kafka topic's replication factor. This defaults to -1 (use the broker default).
`reuse_topic`        | `bool`     | Use the existing Kafka topic after Materialize restarts, instead of creating a new one. The default is false. See [Enabling topic reuse after restart](/sql/create-sink/#exactly-once-sinks-with-topic-reuse-after-restart) for details.
`consistency_topic`  | `text`     | This option is only available to support backwards-compatibility. Please use the new [`CONSISTENCY` syntax](/sql/create-sink/#sink_kafka_connector) to define a consistency topic for the sink.

> `security_protocol`  | `text`     | Use [`ssl`](#ssl-and-authentication) or, for [Kerberos](#ssl-and-authentication), `sasl_plaintext`, `sasl-scram-sha-256`, or `sasl-sha-512` to connect to the Kafka cluster.

Reviewer: ```suggestion
`security_protocol`  | `text`     | Use [`ssl`](#authentication) or, for [Kerberos](#authentication), `sasl_plaintext`, `sasl-scram-sha-256`, or `sasl-sha-512` to connect to the Kafka cluster.
```

`acks`               | `text`     | Sets the number of Kafka replicas that must acknowledge Materialize writes. Accepts values [-1,1000]. `-1` (the default) specifies all replicas.
`retention_ms`       | `long`     | Sets the maximum time Kafka will retain a log.  Accepts values [-1, ...]. `-1` specifics no time limit.  If not set, uses the broker default. {{< version-added v0.9.7 />}}
`retention_bytes`    | `long`     | Sets the maximum size a Kafka partion can grow before removing old logs.  Accepts values [-1, ...]. `-1` specifics no size limit.  If not set, uses the broker default. {{< version-added v0.9.7 />}}
`avro_key_fullname`  | `text`     | Sets the Avro fullname on the generated key schema, if a `KEY` is specified. When used, a value must be specified for `avro_value_fullname`. The default fullname is `row`. {{< version-added v0.18.0 />}}
`avro_value_fullname`| `text`     | Sets the Avro fullname on the generated value schema. When `KEY` is specified, `avro_key_fullname` must additionally be specified. The default fullname is `envelope`. {{< version-added v0.18.0 />}}

> #### SSL and Authentication

Reviewer: ```suggestion
#### Authentication
```

> Sinks use the same ssl and authentication scheme as sources. Please see

Author: is that what a "partial" is? I can try to figure out how that works next week!

> Sinks use the same ssl and authentication scheme as sources. Please see
> [those docs](/sql/create-source/kafka/#authentication) for details and examples/

Reviewer: ```suggestion
Kafka sinks support the same authentication scheme and options as Kafka sources (`SSL`, `SASL`). Check the [Kafka source documentation](/sql/create-source/kafka/#authentication) for more details and examples.
```
